OpenSim/Moco/MocoGoal/MocoContactImpulseTrackingGoal.cpp

```diff
@@ -203,29 +198,27 @@ void MocoImpulseTrackingGoal::initializeOnModelImpl(const Model& model) const {
 
         // Check to see if the model contains a MocoScaleFactor associated with
         // this contact group.
-        std::array<RefPtrMSF, 3> theseScaleFactorRefs;
-        for (int j = 0; j < 3; ++j) {
-            bool foundScaleFactor = false;
-            std::pair<std::string, int> key(group.get_external_force_name(), j);
-            for (const auto& scaleFactor : scaleFactors) {
-                if (m_scaleFactorMap[key] == scaleFactor.getName()) {
-                    theseScaleFactorRefs[j] = &scaleFactor;
-                    foundScaleFactor = true;
-                }
-            }
-            // If we didn't find a MocoScaleFactor for this force direction, set
-            // the reference pointer to null.
-            if (!foundScaleFactor) {
-                theseScaleFactorRefs[j] = nullptr;
+        RefPtrMSF thisScaleFactorRef;
+        bool foundScaleFactor = false;
+        std::pair<std::string, int> key(group.get_external_force_name(), get_impulse_axis());
+        for (const auto& scaleFactor : scaleFactors) {
+            if (m_scaleFactorMap[key] == scaleFactor.getName()) {
+                thisScaleFactorRef = &scaleFactor;
+                foundScaleFactor = true;
             }
         }
-        m_scaleFactorRefs.push_back(theseScaleFactorRefs);
+        // If we didn't find a MocoScaleFactor for this force direction, set
+        // the reference pointer to null.
+        if (!foundScaleFactor) {
+            thisScaleFactorRef = nullptr;
+        }
+        m_scaleFactorRefs.push_back(thisScaleFactorRef);
```

Comment:
For now, until we add support for multiple integrals, we should check that `m_scaleFactorRefs` has length == 1.